### PR TITLE
Dark theme

### PR DESCRIPTION
**Pull request by SeineEloquenz**
This commit adds an automatic dark theme for supported devices by switching to DayNightTheme.

Note: The widget and happy cows layout aren't working in the emulator, they stay light. On connected devices it works fine

image
image

### CODE REVIEW

File: app/src/main/java/com/cypressworks/mensaplan/food/PlanAdapter.java [hunk 1]
```diff
@@ -67,6 +74,9 @@ public View getView(final int position, View v, final ViewGroup parent) {
                 tag = (HeaderHolder) v.getTag();
             }
             final Line line = (Line) items.get(position);
+            if (firstLineName.equals(line.getName())) {
```
Comment (cypressious) [inline app/src/main/java/com/cypressworks/mensaplan/food/PlanAdapter.java]: I think we should do this by checking if `position == 0`

**Comment by cypressious** [comment]
Just the one nitpick, the rest looks good

File: app/src/main/java/com/cypressworks/mensaplan/food/PlanAdapter.java (hunk 1 above)
Comment (SeineEloquenz) [reply app/src/main/java/com/cypressworks/mensaplan/food/PlanAdapter.java]: I did that, but somehow 0 appeared multiple times which led to missing separators in the middle of the list. Commit 804bb86 has this functionality

**Comment by cypressious** [comment]
Did you make the separator visible when the position != 0?


Alexander Linder <notifications@github.com> schrieb am Di., 11. Feb. 2020,
00:18:

**Comment by SeineEloquenz** [comment]
No I did not. I implemented it with ``position == 0`` and the according else branch now and it works.
Could you explain to me why this is necessary? I thought it would suffice to just make it disappear on the first item, but not explicitly make it visible on the others, as I thought the separators would be visible by default

**Comment by cypressious** [comment]
@SeineEloquenz I recommend reading up on view recycling, e.g. here: https://stackoverflow.com/questions/11945563/how-listviews-recycling-mechanism-works

The short version: ListView is reusing a limited number of views to draw all the items. When you scroll, the View that was previously displaying "Linie 1" but is now offscreen will be used to display e.g. "Linie 4". 

Notice, the check `if (v == null)` in the `getView` method? If it is false, it means a previously used view was recycled. 

I _think_ ListView is always reusing views for the same view *type* which is why we need `getItemViewType`. This way a line header is never reused as a meal item.

**Pull request merged**